tests/utils/ExecutorTestBuilder.cpp

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "ExecutorTestBuilder.h"
+#include "ExecutorBenchmarkWrapper.h"
+#include "ExecutorUnitTestWrapper.h"
+#include "TestDeviceManager.h"
+
+using namespace glow;
+using namespace glow::runtime;
+
+void ExecutorTestBuilder::addNode(const std::string &name, DeviceIDTy deviceId,
+                                  llvm::ArrayRef<llvm::StringRef> parents,
+                                  llvm::ArrayRef<llvm::StringRef> inputs,
+                                  llvm::ArrayRef<llvm::StringRef> outputs,
+                                  RunIdentifierTy runId, bool success) {
+  auto newNode = llvm::make_unique<DAGNode>();
+  auto *newNodeRawPtr = newNode.get();
+
+  // If this is the first node being added, record the run ID for the graph.
+  // Otherwise, make sure that the runId matches that of the previous nodes.
+  if (nodes_.empty()) {
+    runId_ = runId;
+  } else {
+    assert(runId == runId_ && "Node run ID does not match rest of graph!");
+  }
+
+  // If the result for this node is false, set the expected
+  // result for the entire test to false.
+  success_ &= success;
```

Comment:
What is the purpose of this bool? Is it to allow tests where running the sub graph fails?

Reply:
Yep.